src/tape/ensemble_frame.py

```diff
@@ -0,0 +1,1136 @@
+            Field name(s) to use as the output frame index. Default is None and index will be
+            inferred from the pandas parquet file metadata, if present. Use False to read all
+            fields as columns.
+        ensemble: `tape.ensemble.Ensemble`, optional
+        |   A link to the Ensemble object that owns this frame.
+        Returns
+        result: `tape.ObjectFrame`
+            The constructed ObjectFrame object.
+        """
+        # Read in the object Parquet file
+        result = dd.read_parquet(
+            path,
+            index=index,
+            columns=columns,
+            split_row_groups=True,
+            engine=TapeObjectArrowEngine,
+        )
+        result.ensemble = ensemble
+        result.label = OBJECT_FRAME_LABEL
+
+        return result
+
+    @classmethod
+    def from_dask_dataframe(cl, df, ensemble=None):
+        """Returns an ObjectFrame constructed from a Dask dataframe..
+        Parameters
+        ----------
+        df: `dask.dataframe.DataFrame` or `list`
+            a Dask dataframe to convert to an ObjectFrame
+        ensemble: `tape.ensemble.Ensemble`, optional
+        |   A link to the Ensemble object that owns this frame.
+        Returns
+        result: `tape.ObjectFrame`
+            The constructed ObjectFrame object.
+        """
+        # Create an ObjectFrame by mapping the partitions to the appropriate meta, TapeObjectFrame
+        # TODO(wbeebe@uw.edu): Determine if there is a better method
+        result = df.map_partitions(TapeObjectFrame)
+        result.ensemble = ensemble
+        result.label = OBJECT_FRAME_LABEL
+        return result
+
+
+"""
+Dask Dataframes are constructed indirectly using method dispatching and inference on the
+underlying data. So to ensure our subclasses behave correctly, we register the methods
+below.
+
+For more information, see https://docs.dask.org/en/latest/dataframe-extend.html
+
+The following should ensure that any Dask Dataframes which use TapeSeries or TapeFrames as their
+underlying data will be resolved as EnsembleFrames or EnsembleSeries as their parrallel
+counterparts. The underlying Dask Dataframe _meta will be a TapeSeries or TapeFrame.
+"""
+get_parallel_type.register(TapeSeries, lambda _: EnsembleSeries)
+get_parallel_type.register(TapeFrame, lambda _: EnsembleFrame)
+get_parallel_type.register(TapeObjectFrame, lambda _: ObjectFrame)
+get_parallel_type.register(TapeSourceFrame, lambda _: SourceFrame)
+
+
+@make_meta_dispatch.register(TapeSeries)
+def make_meta_series(x, index=None):
+    # Create an empty TapeSeries to use as Dask's underlying object meta.
+    result = x.head(0)
+    return result
+
+
+@make_meta_dispatch.register(TapeFrame)
+def make_meta_frame(x, index=None):
+    # Create an empty TapeFrame to use as Dask's underlying object meta.
+    result = x.head(0)
+    return result
+
+
+@meta_nonempty.register(TapeSeries)
+def _nonempty_tapeseries(x, index=None):
```

Comment:
Why do we need it? Should we test it?

Reply:
I think we're lacking in series object testing in general. It makes sense to me that the dispatcher may need to generate a nonempty `TapeSeries` object for `EnsembleSeries` method, but am not 100% confident this is being used. 

Filed https://github.com/lincc-frameworks/tape/issues/315 to note that we should improve testing for workflows involving series as well as whether to remove registering this method.